### DISCUSSION TITLE
Extensions fail to load when using reverse proxy with subpath.

### DISCUSSION THREAD

**Discussion by davidliyutong** [coder/code-server#7006] [Q&A] [answered]
Hello everyone,

I am deploying a code-server instance with Nginx as a reverse proxy. The two are bundled in one docker container. The plan is to serve code-server under `/coder/` subpath, and other services under their unique paths. The editor works fine. I can access the interactive terminal, open every panel, and install extensions from the marketplace. It is just that the extensions are not working.

Here are the details: The code-server and nginx reside in the same container. The code-server is launched with `code-server --bind-addr 0.0.0.0:3000 --auth none --cert false` and listens at port 3000. The nginx daemon listens at port 80. Its configuration
looks like: 

```nginx
location ~ ^/coder/(.*)$ {
        proxy_pass http://127.0.0.1:3000/$1;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "Upgrade";
        proxy_read_timeout 300;
        proxy_set_header Host $http_host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $http_x_forwarded_proto;
        add_header Strict-Transport-Security "max-age=15552000; includeSubDomains" always;
        client_max_body_size 0;
    }
```

The docker container is launched with `docker run --rm -p 3000:3000 -p 8000:80 coderwithnginx:latest`, exposing both port 3000 and port 80. When accessing via `http://127.0.0.1:3000`, the extensions work. However when accessing via `http://127.0.0.1:8000/coder/`, only the editor and terminal work. It looks like the extensions are installed but not activated, and I am getting error messages in the dev console.

> Screenshot for the working scenario
<img width="1742" alt="截屏2024-09-20 13 10 04" src="https://github.com/user-attachments/assets/f18b97ef-97d8-4795-a3e9-6592b48d562b">
> Screenshot for the failed scenario  
<img width="1742" alt="截屏2024-09-20 13 10 37" src="https://github.com/user-attachments/assets/978c34f3-e6b1-4d68-982b-699a475527c9">

What could be the possible causes?

**Comment by code-asher** [comment, marked as answer]
Do the query parameters get included in the location match?  Wonder if you need `proxy_pass http://127.0.0.1:3000/$1$is_args$args;`.

Or this should also work, I think:

```
location /coder/ {
  proxy_pass http://127.0.0.1:3000/;
}
```

**Reply by davidliyutong** [reply]
> Do the query parameters get included in the location match? Wonder if you need proxy_pass http://127.0.0.1:3000/$1$is_args$args;.

This solution works! Thank you very much!
